drivers/builtin/include/mbedtls/pk.h

```diff
@@ -485,6 +341,38 @@ int mbedtls_pk_can_do_ext(const mbedtls_pk_context *ctx, psa_algorithm_t alg,
                           psa_key_usage_t usage);
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
+/**
+ * \brief               Change the PSA algorithm associated with a key.
+ *
+ * \param[in,out] ctx   The PK context to modify.
+ * \param alg           The new policy associated with the key.
+ *                      This may be a PSA algorithm, or a PSA signature
+ *                      algorithm policy where the hash algorithm is
+ *                      the wildcard #PSA_ALG_ANY_HASH.
+ *                      This is the algorithm that
+ *                      mbedtls_pk_get_psa_attributes() will use if it
+ *                      matches the requested usage.
+ *                      This is the algorithm that mbedtls_pk_sign() and
+ *                      mbedtls_pk_verify() will use. These functions will
+ *                      error out if the algorithm is not a signature
+ *                      algorithm.
+ *
+ * \note                This function does not validate that \p alg is
+ *                      compatible with the context. Using an incompatible
+ *                      algorithm may result in an error when trying to
+ *                      use the key, but it will never result in the
+ *                      algorithm being used on a key with an incompatible
+ *                      type.
+ *
+ * \note                The PK module does not enforce key policies.
+ *                      For example, after parsing an ECC key that is
+ *                      designated as ECDH-only through the OID
+ *                      id-ecDH, you can call this function to allow the
+ *                      key to be used for ECDSA (but you probably should not).
+ */
+void mbedtls_pk_set_algorithm(mbedtls_pk_context *ctx,
```

Comment:
This won't work with a future workflow where parsing already creates a PSA key.

I was thinking in the future this might export the key and import it with another policy. But this can fail due to running out of resources. So the function needs to return an error code.

Reply:
On second thoughts, maybe `mbedtls_pk_set_algorithm` shouldn't exist at all. I need to think this further,

Reply:
[Having slept on it again, I don't think `mbedtls_pk_set_algorithm` should exist](https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/203/files#r2009291751).

Action: either here or in a follow-up, implement the copy-based workflow.